src/compiler/checker.ts

```diff
@@ -28515,11 +28515,45 @@ namespace ts {
             }
         }
 
+        function getNonSimpleParameters(parameters: ReadonlyArray<ParameterDeclaration>): ReadonlyArray<ParameterDeclaration> {
+            // ECMA-262 14.1.13
+            if (parameters.length === 0) return [];
+
+            const last = lastOrUndefined(parameters);
+            if (last && isRestParameter(last)) return [last];
+
+            return filter(parameters, parameter => {
+                // ECMA-262 13.3.3.4
+                return !!parameter.initializer || isBindingPattern(parameter.name);
+            });
+        }
+
+        function checkGrammarForUseStrictSimpleParameterList (node: FunctionLikeDeclaration): boolean {
+            const useStrictDirective = node.body && isBlock(node.body) && startsWithUseStrict(node.body.statements) ? first(node.body.statements) : undefined;

```

Comment:
Is it necessary to only do this check if the target is ES2016 or higher? These early errors were introduced in ES2016 and are valid in ES2015: https://www.ecma-international.org/ecma-262/6.0/index.html